File: 4기-코딩테스트-220608/4기 코딩테스트_220608/4기 - 1st 코딩테스트/selectRoom/selectRoom.py
def solution(A):
    answer = 0
    arr = [A[0]]

    while len(arr) != len(A):
        arr.append([])
        for i in range (0, len(arr)):
            l = len(arr) - 1
            if i == 0:
                arr[l].append(A[l][0] + arr[l - 1][0])
            elif i == (len(arr) - 1):
                arr[l].append(A[l][l] + arr[l-1][l-1])
            else:
                arr[l].append(A[l][len(arr[l])] + min(arr[l-1][len(arr[l])], arr[l-1][len(arr[l])-1]))
    return min(arr[len(arr)-1])

File: 4기 코딩테스트_220615/4기 - 1st 코딩테스트/selectRoom/selectRoom.js
function solution(A) {
    var arr = [A[0]];
  
    while(arr.length != A.length){
      arr.push([]);
  
      for(var i = 0; i < arr.length; i++){
        if(i == 0){
          arr[arr.length-1].push(A[arr.length-1][0] + arr[arr.length-2][0]);
        } else if(i == arr.length-1){
          arr[arr.length-1].push(A[arr.length-1][arr.length-1] + arr[arr.length-2][arr.length-2]);
        } else{
          arr[arr.length-1].push(A[arr.length-1][arr[arr.length-1].length]+Math.min(arr[arr.length-2][arr[arr.length-1].length], arr[arr.length-2][arr[arr.length-1].length-1]));
        }
      }
    }
      return Math.min.apply(null, arr[arr.length-1]);
}
  